tests/zfs-tests/tests/functional/cli_root/zfs_destroy/zfs_destroy_008_pos.ksh

```diff
@@ -38,14 +38,27 @@
 #
 ################################################################################
 
+function test_s_run
+{
+	typeset snap=$1
+
+    log_must $ZFS destroy -d $snap
+    log_mustnot datasetexists $snap	
+}
+
 log_assert "'zfs destroy -d <snap>' destroys snapshot if there is no clone"
 log_onexit cleanup_testenv
 
 setup_testenv snap
 
 for snap in $FSSNAP $VOLSNAP; do
-    log_must $ZFS destroy -d $snap
-    log_mustnot datasetexists $snap
+    if [[ $snap == $VOLSNAP ]]; then
+		if is_global_zone; then
+			test_s_run $snap
+		fi
+	else
+		test_s_run $snap
+	fi
```

Comment:
Can you explain what exactly the test failure was and why this is needed?  At the moment under Linux `is_global_zone` should always be true.  The concept of illumos zones hasn't been mapped to Linux style containers for the test suite.  Given that I don't see any functional change here.

However, if one were to run the test suite in a container I could see how that would be a problem.